src/metabase/pulse/render/body.clj

```diff
@@ -242,13 +242,19 @@
 (s/defmethod render :categorical/donut :- common/RenderedPulseCard
   [_ render-type _timezone-id :- (s/maybe s/Str) card {:keys [rows] :as data}]
   (let [[x-axis-rowfn y-axis-rowfn] (common/graphing-column-row-fns card data)
-        rows                        (common/non-nil-rows x-axis-rowfn y-axis-rowfn rows)
-        legend-colors               (zipmap (map (comp str x-axis-rowfn) rows) (cycle colors))
+        rows                        (map (juxt (comp str x-axis-rowfn) y-axis-rowfn)
+                                         (common/non-nil-rows x-axis-rowfn y-axis-rowfn rows))
+        legend-colors               (zipmap (map first rows) (cycle colors))
+        total                       (apply + (map second rows))
+        percentages                 (into {}
+                                          (for [[label value] rows]
+                                            [label (if (zero? total)
+                                                     (tru "N/A")
+                                                     (let [f (DecimalFormat. "###,###.##%")]
```

Comment:
is this compatible with our l10n? unless there's cool magic stuff I missed it seems to hardcode American representation (aka, dot as the decimal separator etc etc)?

Reply:
<img width="1511" alt="image" src="https://user-images.githubusercontent.com/6377293/129801420-462678a9-8c3f-4f1e-bbd1-54ae3872a629.png">
As I understand it from the [javadocs for DecimalFormat](https://docs.oracle.com/javase/7/docs/api/java/text/DecimalFormat.html) the formatting language uses American Locale symbols but there are localized in the implementation.

Reply:
here's Germany:
```
clj -J-Duser.country=DE
Clojure 1.10.3
user=> (.format (java.text.DecimalFormat. "###,###.##%") 123456789.123455)
"12.345.678.912,35%"
user=>
```

Reply:
hey, magic I didn't know about. cool beans